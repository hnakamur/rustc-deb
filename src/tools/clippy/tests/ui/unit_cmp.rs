#![warn(clippy::unit_cmp)]
#![allow(clippy::no_effect, clippy::unnecessary_operation)]

#[derive(PartialEq)]
pub struct ContainsUnit(()); // should be fine

fn main() {
    // this is fine
    if true == false {}

    // this warns
    if {
        true;
    } == {
        false;
    } {}

    if {
        true;
    } > {
        false;
    } {}
}
