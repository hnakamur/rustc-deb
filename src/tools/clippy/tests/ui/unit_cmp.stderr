error: ==-comparison of unit values detected. This will always be true
  --> $DIR/unit_cmp.rs:12:8
   |
LL |       if {
   |  ________^
LL | |         true;
LL | |     } == {
LL | |         false;
LL | |     } {}
   | |_____^
   |
   = note: `-D clippy::unit-cmp` implied by `-D warnings`

error: >-comparison of unit values detected. This will always be false
  --> $DIR/unit_cmp.rs:18:8
   |
LL |       if {
   |  ________^
LL | |         true;
LL | |     } > {
LL | |         false;
LL | |     } {}
   | |_____^

error: aborting due to 2 previous errors

